python/tank/descriptor/io_descriptor/base.py

```diff
@@ -247,8 +249,15 @@ def _find_latest_tag_by_pattern(self, version_numbers, pattern):
             # iterate over versions in list and find latest
             latest_version = None
             for version_number in version_numbers:
-                if is_version_newer(version_number, latest_version):
-                    latest_version = version_number
+                try:
+                    if is_version_newer(version_number, latest_version):
+                        latest_version = version_number
+                except Exception as e:
```

Comment:
Why did you add this new try/except block?

Also, can we have a better exception filtering? Like do we know which type of exception we expect here?